src/Microsoft.Data.SqlClient/tests/FunctionalTests/SqlMetaDataTest.cs

```diff
@@ -2,14 +2,742 @@
+            Assert.Equal("col2", metaData.Name);
+            Assert.Equal(SqlDbType.Decimal, metaData.SqlDbType);
+            Assert.Null(metaData.Type);
+            Assert.Equal(2, metaData.Precision);
+            Assert.Equal(2, metaData.Scale);
+            Assert.True(metaData.UseServerDefault);
+            Assert.True(metaData.IsUniqueKey);
+            Assert.Equal(SortOrder.Ascending, metaData.SortOrder);
+            Assert.Equal(1, metaData.SortOrdinal);
+        }
+
+        [Fact]
+        public void DecimalConstructorWithPrecisionScale()
+        {
+            SqlMetaData metaData = new SqlMetaData("col2", SqlDbType.Decimal, 2, 2);
+            Assert.Equal("col2", metaData.Name);
+            Assert.Equal(SqlDbType.Decimal, metaData.SqlDbType);
+            Assert.Null(metaData.Type);
+            Assert.Equal(2, metaData.Precision);
+            Assert.Equal(2, metaData.Scale);
+        }
+
+        [Fact]
+        public void DecimalConstructorWithNullUdt()
+        {
+            SqlMetaData metaData = new SqlMetaData("col1", SqlDbType.Decimal, 5, 2, 2, 0, SqlCompareOptions.BinarySort, null);
+            Assert.Equal("col1", metaData.Name);
+            Assert.Equal(SqlDbType.Decimal, metaData.SqlDbType);
+            Assert.Equal(5, metaData.MaxLength);
+            Assert.Equal(2, metaData.Precision);
+            Assert.Equal(2, metaData.Scale);
+            Assert.Equal(0, metaData.LocaleId);
+            Assert.Null(metaData.Type);
+        }
+
+        [Fact]
+        public void DecimalConstructorWithPrecisionOutOfRange_Throws()
+        {
+            byte precision = 1;
+            byte scale = 2;
+            ArgumentException ex = Assert.Throws<ArgumentException>(() =>
+            {
+                SqlMetaData metaData = new SqlMetaData("col1", SqlDbType.Decimal, precision, scale, true, true, SortOrder.Ascending, 0);
+            });
+            Assert.NotNull(ex);
+            Assert.NotEmpty(ex.Message);
+            Assert.Contains("precision", ex.Message, StringComparison.OrdinalIgnoreCase);
+        }
+
+        [Fact]
+        public void DecimalConstructorWithPrecisionOutofRange2_Throws()
+        {
+            byte precision = SqlDecimal.MaxPrecision;
+            precision += 1;
+            byte scale = 2;
+            ArgumentException ex = Assert.Throws<ArgumentException>(() =>
+            {
+                SqlMetaData metaData = new SqlMetaData("col1", SqlDbType.Decimal, precision, scale, true, true, SortOrder.Ascending, 0);
+            });
+            Assert.NotNull(ex);
+            Assert.NotEmpty(ex.Message);
+            Assert.Contains("precision", ex.Message, StringComparison.OrdinalIgnoreCase);
+        }
+
+        // TODO: This condition can never bet met because SqlDecimal.MaxPrecision == SqlDecimal.MaxScale
+        // and there's a check that scale cannot exceed precision, so we cannot this exception.
```

Comment:
```suggestion
        // TODO: This condition can never be met because SqlDecimal.MaxPrecision == SqlDecimal.MaxScale
        // and there's a check that scale cannot exceed precision, so we cannot test this exception.
```